Separete DB example from DB reference (#124)
mulesoft/docs-munit: modules/ROOT/pages/db-util.adoc

endif::[]
:page-aliases: munit-database-server.adoc

To test a piece of code that connects with a database server, you must install a database server in your local environment to run the tests. You could also have a dedicated external database server for testing, but your Maven project would not be portable -- a third party would need to install the testing database server to compile your project.

To make it easier for you to test database connections, MUnit allows you to implement a database server in your local environment.

> == Installing the MUnit DB Server Module

Reviewer: DITTO edits made to previous Lines 51-99.
Is this the main "home" for this topic? If so, can we xref to THIS topic as a prerequisite to Configuring Database Server Util in an MUnit Test, rather than as a full subtask?

    <scope>test</scope>
</dependency>
----
+

> The MUnit DB server artifact in your `pom` file must have the `test` scope.

Reviewer: ditto pref edits

</plugin>
----

> == Attributes Reference

Reviewer: This should probably be a definition list rather than a table....

The MUnit DB server is based on H2. Complex structures, index and join commands may not work properly.

[[db-structure-sql]]

> === Defining the DB Structure From an SQL File

Reviewer: "from" in titles is in lowercase, per the Chicago tab within Capitalize My Title (https://capitalizemytitle.com/).
Alas, the tool defaults to APA style rather than Chicago style.... 
(See screen cap I'll send in Slack confirmation of edit being done.)

----

[[db-structure-csv]]

> === Defining the Db Structure From a CSV File

Reviewer: from

Why did you change DB to Db here (and not above)?

</dbserver:config>
----

> You can also split your DB structure among several CSV files. To do this, include the file names as a list separated by a semicolon:

Reviewer: ```suggestion
You can also split your DB structure among several CSV files. To do this, include the file names in a list, with the names separated by semicolons:
```

</dbserver:config>
----

> This creates two tables `FILE_NAME` and `FILE_NAME_1`.

Reviewer: ```suggestion
This creates two tables: `FILE_NAME` and `FILE_NAME_1`.
```

== MUnit DB Server Processors

> The DB util has two processors to interact with your DB server:

Reviewer: Should we be referring to "database utilities" or "DB utilities" rather than "DB util"? Or should DB util be `DB util`?

> === Validate-That

Reviewer: Can we give this value/name a noun object: "Query"?

query="SELECT * FROM jobtitlelookup WHERE JOBTITLE='Developer';" returns="&quot;JOBTITLE&quot;,&quot;EECSALARYORHOURLY&quot;,&quot;JOBTITLEID&quot;\n&quot;Developer&quot;,&quot;10&quot;,&quot;DEV&quot;" />
----

> === Execute

Reviewer: Execute what? is Execute syntax or is it a verb?

> The MUnit DB Server allows you to execute instructions on the in-memory databases, so you can add or remove registries before a test, and also check if your data was stored correctly.

Reviewer: ```suggestion
MUnit DB Server enables you to execute instructions to the in-memory databases so that you can add or remove registries before a test and verify that your data was stored correctly.
```

> Use a DB Server connector with the `execute` operation to insert the values new values to the table (`Culinary Team Member,10,HIR`), then use another DB Server connector with an `execute query` operation to retrieve all values from the table (including the newly added ones) and check the payload using a logger component to store the payload:

Reviewer: DB Server Connector? Or maybe we should be referring to the full name, MUnit DB Server, so that readers don't confuse it, as I apparently have, with some generic database server.
